### ISSUE TITLE
further <graph> issue

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#258]
Following on from #257 if we look at en.wikipedia.org/wiki/Obesity
before
![obesity_bad](https://user-images.githubusercontent.com/861631/47550563-84b86980-d8f7-11e8-9025-910eb50b9459.jpg)
and then with the 'patch' of #257 
![obesity_better](https://user-images.githubusercontent.com/861631/47550686-e7aa0080-d8f7-11e8-988f-631a0b044920.jpg)
In the logs for the session I get an error

    Xoa_utl_svc:page not found: wiki=commons.wikimedia.org page=Data:Sandbox/Doc_James/Obesity_Males_CC-BY-SA.tab

This <graph> is quite different to the 'simple' graphs of say Eatontown,_New_Jersey

Looking at the real en.wikipedia.org it seems to use an image overlay as well as using the data from commons.
I suspect that the logic is not expecting 'data' from commons only 'images'

**Comment by gnosygnu** [comment]
Yeah, in brief, this is probably b/c your commons.wikimedia.org version is from 2017-09 (based on your comment here: #256 ) The obesity page requires at least a version dated 2017-12 or later: see https://commons.wikimedia.org/w/index.php?title=Data:Sandbox/Doc_James/Obesity_Males_CC-BY-SA.tab&action=history

Some other details below.

Thanks!

----

So this is why MediaWiki Graph 2.0 is so painful (as I mentioned in #257). 

V2 now requires that XOWA has access to commons.wikimedia.org. A lot of the .tab data used for the graphs are (a) stored in commons.wikimedia.org and (b) retrieved on the fly through AJAX calls. Incidentally, this is why I abandonned thoughts of getting this working in Html dumps as it would require a lot of additional work to get this working on Android. 

I could do a workaround by adding the graph-image-overlay to the general image download process. However, this meant adding more code (figure out the proper file-name based on the hash of the Graph node contents; add it to the XOWA download process) and would bloat the image databases more. Also, I liked the idea of having things as editable as possible, so really wanted to get the Graph extension working, rather than cop-out and just download the image.

Anyway, just in case you're curious...
